### PR TITLE
feat(behavior_path_planner): add option for combining arc pull out paths

### PR DESCRIPTION
**Pull request by kosuke55**
Signed-off-by: kosuke55 <kosuke.tnp@gmail.com>

## Description

Add option for combining two arc paths of pull_out. This eliminates stops along the way.

https://user-images.githubusercontent.com/39142679/212863730-41b8c057-ecb0-408b-b636-76ad8877efdb.mp4

## Related links

tier4 internal link 
- https://star4.slack.com/archives/CEV8XMJBV/p1673857697901869

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/include/behavior_path_planner/scene_module/pull_out/pull_out_parameters.hpp [hunk 1]
```diff
@@ -38,6 +38,7 @@ struct PullOutParameters
   double deceleration_interval;
   // geometric pull out
   bool enable_geometric_pull_out;
+  bool enable_combined_path;
```
Comment (tkimura4) [inline planning/behavior_path_planner/include/behavior_path_planner/scene_module/pull_out/pull_out_parameters.hpp:41]: This parameter name is a little confusing.
(path is usually combined. )

Comment (tkimura4) [reply planning/behavior_path_planner/include/behavior_path_planner/scene_module/pull_out/pull_out_parameters.hpp:41]: I think a name like ```divide_pull_out_path``` is easier to understand

Comment (kosuke55) [reply planning/behavior_path_planner/include/behavior_path_planner/scene_module/pull_out/pull_out_parameters.hpp]: thanks, updated the name in https://github.com/autowarefoundation/autoware.universe/pull/2669/commits/18ce9aff21cf416db9edd1495528f53aa82c481c

**Review by tkimura4 — APPROVED** [review]
https://user-images.githubusercontent.com/59680180/212874385-926ed9e4-1e16-4f75-b74f-ad12452f72db.mp4

LGTM

**Pull request merged**